lightning/src/chain/chainmonitor.rs

```diff
@@ -656,6 +685,26 @@ where C::Target: chain::Filter,
 			}
 		}
 	}
+
+	/// Archives channel monitors which are stale, meaning they have
+	/// no funds to claim and are eligible for removal. 
+	///
+	/// This is useful for pruning stale channels from the monitor set.
+	///
+	/// The monitor data is archived to a backup location and then removed from the primary storage.
+	pub fn archive_stale_channel_monitors(&self, to_archive: Vec<OutPoint>) {
+		let mut monitors = self.monitors.write().unwrap();
+		for funding_txo in to_archive {
+			let channel_monitor = monitors.get(&funding_txo);
+			if let Some(channel_monitor) = channel_monitor {
+				if channel_monitor.monitor.is_stale() 
```

Comment:
this if is safe because rust goes through it LTR, so the channel will be archived only if its stale, and it will be removed only if it was archived